harvester/harvest.py

```diff
@@ -57,6 +58,126 @@
 
 ROOT_DIR = Path(__file__).parents[1]
 
+# harvest worker count
+harvest_worker_sync_count = int(os.getenv("HARVEST_WORKER_SYNC_COUNT"))
+
+
+class CKANSyncTool:
+    """A helper class used for parallelization of CKAN network calls.
+
+    Usage:
+        ckan_sync_tool = CKANSyncTool()
+        ckan_sync_tool.sync(record)
+    """
+
+    def sync(self, record):
+        """General sync bus for all records to pass through.
+
+        Args:
+            self: CKANSyncTool class
+            record: An instance of HarvestRecord (Record) class
+
+        Returns:
+            True (bool): on record success
+
+        Raises:
+            DCATUSToCKANException: Any DCAT-US to CKAN metadata transform issue
+            SynchronizeException: Any ckan sync issue
+        """
+        if record.status == "success" or record.status == "error":
+            logger.info(
+                f"{record.identifier} has status '{record.status}'. bypassing {record.action}"
+            )
+            return
+
+        # pre-sync try to ckanify record metadata
+        try:
+            if record.action == "create" or record.action == "update":
+                self.ckanify_record(record)
+        except Exception as e:
+            record.status = "error"
+            raise DCATUSToCKANException(
+                repr(e), record.harvest_source.job_id, record.id
+            )
+
+        # begin ckanpi sync proper
+        start = get_datetime()
+        try:
+            if record.action == "delete":
+                self.delete_record(record)
+            elif record.action == "create":
+                self.ckanify_record(record)
```

Comment:
question (non-blocking)**: Didn't we just do this  in the pre-sync step? Is there some reason we need to do it again?

Reply:
I deleted the pre_sync and combined them back together.